3. copy-vs-reference/index.js

```diff
@@ -116,16 +137,17 @@ let user3 = {
   house: "Stark",
   brothers: ["John", "Bran", "Robb"]
 };
+
 // Output of the below code and why?
-user.house === user2.house; // output:
-user.house == user2.house; // output:
-user.brothers === user2.brothers; // output:
-user.brothers == user2.brothers; // output:
-user.name == user2.name; // output:
-user.name === user2.name; // output:
-user.brothers == user3.brothers; // output:
-user.brothers === user3.brothers; // output:
-user.house === user2.house; // output
-user.house === user3.house; // output
-user.brothers[0] === user2.brothers[0]; // output
-user.brothers[0] === user3.brothers[0]; // output
+user.house === user2.house; // output: true, as both having same house.
+user.house == user2.house; // output: true, as both having same house.
+user.brothers === user2.brothers; // output: true, as both point to same array.
+user.brothers == user2.brothers; // output: true, as both point to same array.
+user.name == user2.name; // output: true, both are same.
+user.name === user2.name; // output: true, both are same.
+user.brothers == user3.brothers; // output: true, both are same.
+user.brothers === user3.brothers; // output: true, both are same.
+user.house === user2.house; // output true, both are same.
+user.house === user3.house; // output true, both are same.
+user.brothers[0] === user2.brothers[0]; // output true, both are pointing to same array.
+user.brothers[0] === user3.brothers[0]; // output true, both are pointing to same array.
```

Comment:
Do the whole thing again. Keeping in mind `primitive` copy by value, `non-primitive` copy by reference.

Reply:
Correction done at line 142 to 153.